jsettlers.logic/src/main/java/jsettlers/logic/movable/Movable.java

```diff
@@ -973,4 +1097,90 @@ public String toString() {
 		DEBUG_STATE
 	}
 
+	public void increaseStateProgress(float step) {
+		this.constructionProgress += step;
+	}
+
+	public boolean isShip() {
+		return ships.contains(this.movableType);
+	}
+
+	public final ImageLink[] getImages() {
+		return images;
+	}
+
+	public final ImageLink[] getBuildImages() {
+		return buildImages;
+	}
+
+	public final void setDirection(EDirection direction) {
+		this.direction = direction;
+	}
+
+	public boolean addPassenger(Movable movable) {
+		if (passengers.size() < 7) {
+			this.passengers.add(movable);
+			return true;
+		}
+		return false;
+	}
+
+	public ArrayList<IMovable> getPassengers() {
+		return this.passengers;
+	}
+
+	public boolean unload(ShortPoint2D position) {
+		if (this.getMovableType() != EMovableType.FERRY) {
+			return false;
+		}
+		if (position == null) {
+			return false;
+		}
+		this.unloadingPosition = position;
+		if (this.passengers.size() > 0) {
+			setState(EMovableState.UNLOADING);
+		}
+		return true;
+	}
+
+	private boolean checkStackNumber(int stack) {
+		return stack >= 0 && stack < this.cargoStacks;
+	}
+
+	public void setCargoType(EMaterialType cargo, int stack) {
+		if (checkStackNumber(stack)) {
```

Comment:
Throw an `IllegalArgumentException` otherwise. This should not happen and it is important that we find the cause for this if it does.